vispy/visuals/gridlines.py

```diff
@@ -22,6 +27,8 @@
     local_pos /= local_pos.w;
     dx = dx / dx.w - local_pos;
     dy = dy / dy.w - local_pos;
+    if (px_pos.z > 1 || px_pos.z < -1)
+        discard;
```

Comment:
Could you add a comment to this? Not immediately obvious to me why it's needed.

Reply:
The grid should be planar, otherwise, it will induce artifacts

Reply:
Unless I'm missing something the `px_pos` will be the pixel position on the Canvas/screen. `z` should never be anything except 0, right? Or is it possible for `z` to not be `0` if some other transforms shift the `z` level of the Visual...even so that wouldn't necessarily mean that the grid isn't planar. Sorry if I'm missing something.